reference-controllers/rabbitmq-controller/src/main.rs

```diff
@@ -237,49 +237,16 @@ async fn restart_sts_if_needed(rabbitmq: &RabbitmqCluster, client: &Client) -> R
     );
```

Comment:
Do we need a separate `restart_sts_if_needed` or we can just merge it into the branch that updates the sts inside `reconcile_stateful_set`?